src/api/mod.rs

```
@@ -175,13 +185,14 @@ fn process(handle: Handle, socket: UnixStream) {
     tokio::spawn(future);
 }
 
-pub fn run(handle: Handle) -> Result<()> {
-    let p = path::Path::new("/var/run");
-    fs::create_dir_all(p)?;
-    let p = p.join(SOCKET_NAME);
-    let _ = fs::remove_file(&p);
+pub fn run(handle: Handle) -> PathResult<()> {
+    let path = Path::new("/var/run");
+    fs::create_dir_all(path).context(Metadata { path })?;
+    let p = path.join(SOCKET_NAME);
+    fs::remove_file(&p).context(Metadata { path: &p })?;
```

Actually it was intended to ignore the error from remove_file, because wed don't care if the file does not exist!